wordpress/wp-content/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+class Webonary_Cloud
+{
+	public static $doBrowseByLetter = 'browse';
+
+	public static $doGetEntry = 'get';
+
+	public static $doSearchFulltext = 'search';
+
+	private static function isValidEntry($entry) {
+		return is_object($entry) && isset($entry->_id);
+	} 
+
+	private static function convertGuidToId($guid) {
+		return 'g' . $guid;
+	} 
+
+	private static function remoteGet($request, $args = array()) {
+		if (!defined('WEBONARY_CLOUD_API_URL'))  {
+			error_log('WEBONARY_CLOUD_API_URL is not set! Please do so in wp-config.php.');
+			return;
+		}
+
+		$url = WEBONARY_CLOUD_API_URL . $request;
+
+		if (WP_DEBUG){
+			error_log('Getting results from ' . $url);
+			if (count($args)) {
+				error_log('using ' . print_r($args, true));
+			} 
+		}
+
+		return wp_remote_get($url, $args);
+	}
+
+	private static function remoteFileUrl($filePath) {
+		if (defined('WEBONARY_CLOUD_FILE_URL')) {
+			$fileUrl = WEBONARY_CLOUD_FILE_URL;
+		}
+		else {
+			error_log('WEBONARY_CLOUD_FILE_URL is not set! Please do so in wp-config.php.');
+			$fileUrl = '';
+		}
+		return $fileUrl . $filePath; 
+	}
+
+	public static function entryToFakePost($dictionary, $entry) {	
```

Comment:
Code like this is one of the reasons I prefer to not use WordPress. There is no way for a reviewer to know if this produces valid HTML.

OK, the rant is over now.

Reply:
You will like React more :-)